README.md

```diff
@@ -85,35 +86,50 @@ imgix.request(`assets/${sourceId}/${originPath}`)
 .then(response => console.log(JSON.stringify(response, null, 2)));
 ```
 
-### Patch asset metadata
+### Retrieve custom fields for a single image
 
 ```js
-var metadata = {
-    "data": {
-        "attributes": {
-            "categories": [
-                "Dessert"
-            ],
-            "labels_user_defined": {
-                "Pecan pie": 0
-            },
-            "metadata": {
-                "my_custom_key": "my_custom_value"
-            }
-        },
-        "id": `${sourceId}/uploads/pecanpie.jpg`,
-        "type": "assets"
+var customFields;
+
+imgix.request(`assets/${sourceId}/uploads/pecanpie.jpg`)
+.then(response => {
+    customFields = response.data.attributes.customFields;
+    console.log(customFields);
+});
+```
+
+### Update custom fields for a single image
+
+```js
+var document = {
+    'data': {
+        'attributes': {}
     }
 };
 
-imgix.request('assets/${sourceId}/uploads/pecanpie.jpg',{
-    method: 'PATCH',
-    body: metadata
+ix.request(`assets/${sourceId}/uploads/pecanpie.jpg`)
+.then(response => {
+    /*
+    ** Populate `document` with all pre-existing fields
+    ** so as to not overwrite them when sending a PATCH request
+    **
+    ** `id` and `type` must be included in the request as well
+    */
+    document.data.attributes.custom_fields = response.data.attributes.custom_fields;
+    document.data.attributes.custom_fields.type = 'dessert';
+    document.data.id = response.data.id;
+    document.data.type = response.data.type;
```

Comment:
I think defining these two attributes here kind of muddies the waters a bit. In a real-world scenario, you'd know the ID and type of the object you're planning to patch, so why not just include them up on L104-108 where the `document` structure is being built?

Reply:
Out of curiosity, can `type` ever be anything other than `assets`?

Reply:
If you're hitting a non-asset endpoint, yes! Sources use `type: "sources"`, users use `type: "users"`, etc. etc.